python/ray/serve/application.py

```diff
@@ -0,0 +1,190 @@
+        self.deploy()
+
+        logger.info("\nDeployed successfully!\n")
+
+        while True:
+            statuses = serve_application_status_to_schema(self.get_statuses()).json(
+                indent=4
+            )
+            logger.info(f"{statuses}")
+            time.sleep(10)
+
+    def to_json(self, f: Optional[TextIO] = None) -> str:
+        """Write list of deployments to json str or file."""
+
+        json_str = serve_application_to_schema(self._deployments).json(indent=4)
+
+        if f:
+            f.write(json_str)
+        return json_str
+
+    @classmethod
+    def from_json(cls, str_or_file: Union[str, TextIO]) -> "Application":
+        """Load list of deployments from json str or file."""
+
+        if isinstance(str_or_file, str):
+            schema = ServeApplicationSchema.parse_raw(
+                str_or_file, content_type="application/json"
+            )
+        else:
+            schema = ServeApplicationSchema.parse_obj(json.load(str_or_file))
+
+        return cls(schema_to_serve_application(schema))
+
+    def to_yaml(self, f: Optional[TextIO]) -> Optional[str]:
+        """Write list of deployments to yaml str or file."""
+
+        json_str = serve_application_to_schema(self._deployments).json(indent=4)
+
+        if f:
+            yaml.safe_dump(json_str, f)
+        return yaml.safe_dump(json_str)
+
+    @classmethod
+    def from_yaml(cls, str_or_file: Union[str, TextIO]) -> "Application":
+        """Load list of deployments from yaml str or file."""
+
+        deployments_json = yaml.safe_load(str_or_file)
+        schema = ServeApplicationSchema.parse_obj(deployments_json)
+        return cls(schema_to_serve_application(schema))
+
+    def _configure_runtime_env(self, deployment: Deployment, updates: Dict):
+        """
+        Overwrites deployment's runtime_env with fields in updates. Any fields
+        in deployment's runtime_env that aren't in updates stay the same.
+        """
+
+        if deployment.ray_actor_options is None:
+            deployment._ray_actor_options = {"runtime_env": updates}
+        elif "runtime_env" in deployment.ray_actor_options:
+            deployment.ray_actor_options["runtime_env"].update(updates)
+        else:
+            deployment.ray_actor_options["runtime_env"] = updates
+
+    def __getitem__(self, key: str):
+        """Fetch deployment by name using dict syntax: app["name"]"""
+
+        if key in self._deployments:
+            return self._deployments[key]
+        else:
+            raise KeyError(
+                "Serve application does not contain a " f'"{key}" deployment.'
+            )
+
+    def __getattr__(self, name: str):
+        """Fetch deployment by name using attributes: app.name"""
+
+        if name in self._deployments:
+            return self._deployments[name]
+        else:
+            raise AttributeError(
+                "Serve application does not contain a " f'"{name}" deployment.'
```

Comment:
is this re-formatted to the same line by accident ?

Reply:
Yes, good catch. I fixed it.